README.md

```diff
@@ -1,16 +1,644 @@
+  using  URI 
+  do     Ability
+  inputs Any
+} 
+```
+
+## 3.3 JSON Examples
+
+### 3.3.1 Run All
+
+ ``` js
+{
+  "ucan/invoke": "QmYW8Z58V1v8R25USVPUuFHtU7nGouApdGTk3vRPXmVHPR",
+  "v": "0.1.0",
+  "nnc": "abcdef",
+  "ext": null,
+  "sig": "bdNVZn_uTrQ8bgq5LocO2y3gqIyuEtvYWRUH9YT-SRK6v_SX8bjt-VZ9JIPVTdxkWb6nhVKBt6JGpgnjABpOCA"
+}
+```
+
+ ``` js
+{
+  "ucan/invoke": "QmYW8Z58V1v8R25USVPUuFHtU7nGouApdGTk3vRPXmVHPR",
+  "v": "0.1.0",
+  "nnc": "abcdef",
+  "run": "*", // Explicitly "run all"
+  "ext": null,
+  "sig": "bdNVZn_uTrQ8bgq5LocO2y3gqIyuEtvYWRUH9YT-SRK6v_SX8bjt-VZ9JIPVTdxkWb6nhVKBt6JGpgnjABpOCA"
+}
+```
+
+### 3.3.2 Promise Pipelines
+
+Promise pipelines are handled in more detail in [section 5](#5-promise-pipelining). In brief, they enable taking the output of an action that has not yet run as the input to another action. Here is a simple example:
+
+``` js
+{
+  "ucan/invoke": "QmY4jEVE35u8SHegWoDak2x6vTAZ6Cc4cpSD5LAUQ23W7L",
+  "v": "0.1.0",
+  "nnc": "02468",
+  "ext": null,
+  "run": {
+    "notify-bob": {
+      "using": "mailto://alice@example.com",
+      "do": "msg/send",
+      "inputs": [
+        {
+          "to": "bob@example.com",
+          "subject": "DNSLink for example.com",
+          "body": "Hello Bob!"
+        }
+      ]
+    },
+    "log-as-done": {
+      "using": "https://example.com/report"
+      "do": "crud/update"
+      "inputs": {
+        "from": "mailto://alice@exmaple.com",
+        "to": ["bob@exmaple.com"],
+        "event": "email-notification",
+        "value": {"ucan/promise": ["/", "notify-bob"]} // Pipelined promise
+      }
+    }
+  },
+  "sig": "5vNn4--uTeGk_vayyPuNTYJ71Yr2nWkc6AkTv1QPWSgetpsu8SHegWoDakPVTdxkWb6nhVKAz6JdpgnjABppC7"
+}
+```
+
+# 4 Receipt
+
+An invocation receipt is an attestation of the output of an invocation. A receipt MUST be signed by the executor (the `aud` of the associated UCAN).
+
+Note that this does not guarantee correctness of the result! The statement's veracity MUST be only understood as an attestation from the executor.
+
+## 4.1 Fields
+
+| Field          | Type         | Description                                                                                      | Required | Default |
+|----------------|--------------|--------------------------------------------------------------------------------------------------|----------|---------|
+| `ucan/receipt` | `CID`        | CID of the Invocation that generated this response                                               | Yes      |         |
+| `rlt`          | `{CID: Any}` | The results of each call, indexed by the CID of the `dag-cbor` encoded [Action](#32-ipld-schema) | Yes      |         |
+| `v`            | `SemVer`     | SemVer of the UCAN invocation object schema                                                      | Yes      |         |
+| `nnc`          | `String`     | A unique nonce, to distinguish each receipt                                                      | Yes      |         |
```

Comment:
Can same invocation have multiple receipts ? Unless so would not `ucan/receipt` already guarantee uniqueness ?